### PR TITLE
Fixes Windows Authentication in GitAuthorizeAttribute

### PR DESCRIPTION
**Pull request by los93sol**
This change fixes Windows authentication so that when it is enabled the user is not prompted for credentials, instead they are passed via Basic authentication.

In addition, this change modifies the user check to pass their credentials with the domain the same way the check happens from the web interface.

### CODE REVIEW

File: Bonobo.Git.Server/Attributes/GitAuthorizeAttribute.cs [hunk 1]
```diff
@@ -67,11 +69,25 @@ public override void OnAuthorization(System.Web.Mvc.AuthorizationContext filterC
...
-                    filterContext.Result = new HttpStatusCodeResult(HttpStatusCode.Unauthorized);
-                    return;
+                    if (AuthenticationProvider is WindowsAuthenticationProvider)
+                    {
+                        AuthenticationProperties authenticationProperties = new AuthenticationProperties()
+                        {
+                            RedirectUri = httpContext.Request.Path
+                        };
+
+                        httpContext.Request.GetOwinContext().Authentication.Challenge(authenticationProperties, WindowsAuthenticationDefaults.AuthenticationType);
```
Comment (RedX2501) [inline Bonobo.Git.Server/Attributes/GitAuthorizeAttribute.cs]: This means that after Basic Auth their git client will try their username and password from windows? If that fails it'll ask the user for username and password?

**Comment by larshg** [comment]
I have been testing this lately and I can't get it work with NTLM - also according to [this] post git doesn't support NTLM - but only Basic auth.
However you can set up proxy things inbetween. Or you can use the git credential manager to save passwords to avoid logging in more than once.

**Comment by larshg** [comment]
@los93sol I'm closing this as it doesn't seem to correct anything that would actually work? If you think otherwise please reopen and supply additional information.

**Comment by larshg** [comment]
I just need to test this a bit more after I have setup Git credential manager to handle windows authentication - then this might be necessary... At least some of it!

**Comment by larshg** [comment]
I tried it again, but it keeps returning 401 and enters UnauthorizedResult in gitcontroller ?

**Comment by larshg** [comment]
Closing due to inactivity

**Pull request closed without merging**